presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -434,8 +442,15 @@ public WriteMapping toWriteMapping(ConnectorSession session, Type type)
         if (TIMESTAMP_MILLIS.equals(type)) {
             return WriteMapping.longMapping("timestamp", timestampWriteFunction());
         }
-        if (TIMESTAMP_WITH_TIME_ZONE.equals(type)) {
-            return WriteMapping.longMapping("timestamptz", timestampWithTimeZoneWriteFunction());
+        if (type instanceof TimestampWithTimeZoneType && ((TimestampWithTimeZoneType) type).getPrecision() <= 6) {
```

Comment:
Until 9 we would be fine. It would harder with 10,11,12 as I am not sure if that is compatible with JDBC. At least not via `OffsetDateTime` PSQL uses.

The `((TimestampWithTimeZoneType) type).getPrecision() <= 6` condition is merly a checkstate. I just did not want to have it explicitly in `if (type instanceof TimestampWithTimeZoneType) {`, so standard unsupported type handling logic triggers if for some magical reason we get timestamp with precision >6 (e.g with some future version of PSQL)